packages/nuxt/src/app/nuxt.ts

```diff
@@ -458,31 +464,31 @@ export function callWithNuxt<T extends (...args: any[]) => any> (nuxt: NuxtApp |
 
 /* @__NO_SIDE_EFFECTS__ */
 /**
- * Returns the current Nuxt instance.
+ * Returns the Nuxt instance from the given app name if provided.
  *
  * Returns `null` if Nuxt instance is unavailable.
  * @since 3.10.0
  */
-export function tryUseNuxtApp (): NuxtApp | null {
+export function tryUseNuxtApp (appName?: string): NuxtApp | null {
   let nuxtAppInstance
   if (hasInjectionContext()) {
     nuxtAppInstance = getCurrentInstance()?.appContext.app.$nuxt
   }
 
-  nuxtAppInstance = nuxtAppInstance || nuxtAppCtx.tryUse()
+  nuxtAppInstance = nuxtAppInstance || getNuxtAppCtx(appName).tryUse()
 
   return nuxtAppInstance || null
 }
 
 /* @__NO_SIDE_EFFECTS__ */
 /**
- * Returns the current Nuxt instance.
+ * Returns the Nuxt instance from the given app name if provided.
  *
  * Throws an error if Nuxt instance is unavailable.
  * @since 3.0.0
  */
-export function useNuxtApp (): NuxtApp {
-  const nuxtAppInstance = tryUseNuxtApp()
+export function useNuxtApp (appName?: string): NuxtApp {
```

Comment:
From the outside, `appName` might make more sense than `buildId` but as it's defaulting to `buildId` for now, should we actually use `buildId`?

Reply:
I think `appName` makes sense here 👍